onboard/nav/avoidances.cpp

```diff
@@ -0,0 +1,164 @@
+/* Searches.cpp */
```

Comment:
Be careful copy and pasting code.
Also, please make this file only for the one obstacle avoidance algorithm. We will make more files when we make a new algorithm. This will help with conflicts in development. Ask David if you don't believe me as he suffered before I learned my lesson here.

Otherwise the logic looks generally good for the PR.